<a href="https://colab.research.google.com/github/mildredmiranda16/mildredmiranda16.github.io/blob/main/Art%C3%ADculo_t%C3%A9cnico.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Artículo Técnico
## **Análisis bayesiano sobre la adquisición de seguros para autos en EUA ***

Alumna: Mildred Miranda Anguas

In [ ]:
# Llamar las librerias conocidas que utilizaremos
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import colors
import seaborn as sb

%matplotlib inline
plt.rcParams['figure.figsize'] = (16, 9)
plt.style.use('ggplot')

In [ ]:
# Para el modelo predictivo utilizaremos la libreria scikit learn
# Así que cargaremos las librerias

from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix
from sklearn.naive_bayes import GaussianNB
from sklearn.feature_selection import SelectKBest


In [ ]:
#Leer el data set
basecar = pd.read_excel("/content/Mildred Miranda Anguas - insurance_challenge_train.xlsx", sheet_name= "Sheet1")


In [ ]:
# Modificar la columna APERSAUT: 0 se queda igual, del 1 al 7 se convierte en 1
basecar['APERSAUT'] = basecar['APERSAUT'].apply(lambda x: 0 if x == 0 else 1)

# Modificar la columna MZPART: 0 se queda igual, del 1 al 9 se convierte en 1
basecar['MZPART'] = basecar['MZPART'].apply(lambda x: 0 if x == 0 else 1)

# Modificar la columna MFWEKIND: 0 se queda igual, del 1 en adelante se convierte en 1
basecar['MFWEKIND'] = basecar['MFWEKIND'].apply(lambda x: 0 if x == 0 else 1)


In [ ]:
# Seleccionar solo las columnas especificadas
colsel = ['APERSAUT', 'MZPART', 'MFWEKIND']
tabla = basecar[colsel]

# Mostrar la tabla resultante
print(tabla.head(10)) #Ver las primeras 10 lineas de la nueva tabla

   APERSAUT  MZPART  MFWEKIND
0         1       1         1
1         0       1         1
2         1       0         1
3         1       1         1
4         0       1         1
5         1       0         1
6         1       0         1
7         0       1         1
8         1       1         1
9         0       1         1


In [ ]:
tabla.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5822 entries, 0 to 5821
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   APERSAUT  5822 non-null   int64
 1   MZPART    5822 non-null   int64
 2   MFWEKIND  5822 non-null   int64
dtypes: int64(3)
memory usage: 136.6 KB


In [ ]:
# Verificar los cambios
print(tabla['APERSAUT'].value_counts())
print(tabla['MZPART'].value_counts())
print(basecar['MFWEKIND'].value_counts())

APERSAUT
1    2977
0    2845
Name: count, dtype: int64
MZPART
1    4970
0     852
Name: count, dtype: int64
MFWEKIND
1    5669
0     153
Name: count, dtype: int64


De las 5822 observaciones, 2977 tienen alguna poliza de seguro mientras que 2845 no tienen. Por otra parte, del total de observaciones, 4970 tienen seguro privado, mientras que 852 hogares no cuentan con seguro privado. Por ultimo, 5669 de los hogares tienen hijos y solo 153 no tienen

In [ ]:
#Divido las variables etiquetas
X=tabla.drop(['APERSAUT'], axis=1)
y=tabla['APERSAUT']# target o objetivo

In [ ]:
# Split dataset in training and test datasets
X_train, X_test = train_test_split(tabla, test_size=0.2, random_state=6)
y_train =X_train["APERSAUT"]
y_test = X_test["APERSAUT"]

In [ ]:
X_train.head()

APERSAUT  MZPART  MFWEKIND
3306         0       1         1
1618         1       1         1
3074         0       0         1
5388         0       1         1
835          1       1         1

In [ ]:
tablax = ['MZPART', 'MFWEKIND']
X_train[tablax].head()

MZPART  MFWEKIND
3306       1         1
1618       1         1
3074       0         1
5388       1         1
835        1         1

In [ ]:
from sklearn.naive_bayes import GaussianNB

# objeto del modelo
gnb = GaussianNB()
# entranar el modelo
gnb.fit(X_train[tablax].values,y_train)

# Hacemos las prediciones
y_pred = gnb.predict(X_test[tablax])

print('Precisión en el set de Entrenamiento: {:.2f}'
     .format(gnb.score(X_train[tablax], y_train)))
print('Precisión en el set de Test: {:.2f}'
     .format(gnb.score(X_test[tablax], y_test)))

Precisión en el set de Entrenamiento: 0.51
Precisión en el set de Test: 0.51


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:486: UserWarning: X has feature names, but GaussianNB was fitted without feature names
  warnings.warn(
/usr/local/lib/python3.10/dist-packages/sklearn/base.py:486: UserWarning: X has feature names, but GaussianNB was fitted without feature names
  warnings.warn(
/usr/local/lib/python3.10/dist-packages/sklearn/base.py:486: UserWarning: X has feature names, but GaussianNB was fitted without feature names
  warnings.warn(
